src/main/kotlin/sc/iview/commands/demo/advanced/InstancingBenchmark.kt

```diff
@@ -0,0 +1,217 @@
+import org.joml.Vector3f
+import org.joml.Vector4f
+import org.scijava.ItemVisibility
+import org.scijava.command.Command
+import org.scijava.command.CommandService
+import org.scijava.log.LogService
+import org.scijava.plugin.Menu
+import org.scijava.plugin.Parameter
+import org.scijava.plugin.Plugin
+import sc.iview.SciView
+import sc.iview.SciView.Companion.create
+import sc.iview.commands.MenuWeights
+import sc.iview.commands.demo.animation.ParticleDemo
+import java.util.*
+import kotlin.streams.asStream
+
+/**
+ * A benchmark that compares the conventional node publishing method with instanced geometry.
+ * Nodes can either be published separately or cumulatively at the end.
+ * Creating the instanced geometry can be run sequentially or in parallel (using a stream over an array).
+ *
+ * @author Samuel Pantze
+ */
+
+@Plugin(
+    type = Command::class,
+    label = "Instancing Benchmark",
+    menuRoot = "SciView",
+    menu = [Menu(label = "Demo", weight = MenuWeights.DEMO), Menu(
+        label = "Advanced",
+        weight = MenuWeights.DEMO_ADVANCED
+    ), Menu(label = "Instance Benchmark", weight = MenuWeights.DEMO_ADVANCED_INSTANCE_BENCHMARK)]
+)
+class InstancingBenchmark : Command {
+    @Parameter
+    private lateinit var log: LogService
+
+    @Parameter
+    private lateinit var sciview: SciView
+
+    @Parameter(persist = false, label = "Number of instances", min = "1", max = "100000000", stepSize = "1")
+    private var numParticles = System.getProperty("sciview.benchmark.numParticles")?.toInt() ?: 1_000
+
+    @Parameter(visibility = ItemVisibility.MESSAGE)
+    private var message = "<html> <body>" +
+            "    <div> The following execution times can be expected for each</div>" +
+            "    <div> run type for 1000 instances: </div>" +
+            "    <div> - publishAll:  ~ 20 seconds </div>" +
+            "    <div> - publishOnce: ~ 1 second </div>" +
+            "    <div> - instanceSequential:  ~ 0.1 seconds </div>" +
+            "    <div> - instanceParallel:    &lt; 0.1 seconds </div>" +
+            "</body> </html>"
+
+    @Parameter(persist = false, choices = ["publishAll", "publishOnce", "instanceSequential", "instanceParallel"], label = "Type of benchmark")
+    private var benchmarkType = System.getProperty("sciview.benchmark.type") ?: "publishAll"
+
+    val rng = Random(17)
+    @Parameter(label = "X range", min = "0.1", max = "1000", style = "format:0.0", stepSize = "0.1")
+    private var maxX = 10f
+    @Parameter(label = "Y range", min = "0.1", max = "1000", style = "format:0.0", stepSize = "0.1")
+    private var maxY = 10f
+    @Parameter(label = "Z range", min = "0.1", max = "1000", style = "format:0.0", stepSize = "0.1")
+    private var maxZ = 10f
+
+
+    override fun run() {
+
+        var ranSuccessfully = true;
+        log.info("Running instancing benchmark with $numParticles agents and type \"$benchmarkType\".")
+        val startTime = System.currentTimeMillis()
```

Comment:
Kotlin has helper functions for measuring timing, see https://kotlinlang.org/docs/time-measurement.html#measure-time 

In addition, when not using Kotlin's helpers, use System.nanoTime(), which uses the system's high-resolution timer (for more details see e.g. [this thread](https://stackoverflow.com/questions/351565/system-currenttimemillis-vs-system-nanotime) on StackOverflow)